#!/usr/bin/env python3
PACKAGE = "follow_lane_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
                        # level,       defaut, min, max

gen.add("thresh", int_t, 0, "Thresh Value", 250, 0, 255)

gen.add("inner_lane", bool_t, 0, "Inner", False)
gen.add("outer_lane", bool_t, 0, "Outer", False)

gen.add("hue_l", double_t, 0, "hue low",  0, 0, 179)
gen.add("hue_h", double_t, 0, "hue high", 130, 0, 179)

gen.add("sat_l", double_t, 0, "sat low",  60,   0, 255)
gen.add("sat_h", double_t, 0, "sat high", 255, 0, 255)

gen.add("val_l", double_t, 0, "val low",  70,   0, 255)
gen.add("val_h", double_t, 0, "val high", 255, 0, 255)

exit(gen.generate(PACKAGE, "detect_red", "DetectRed"))
# The 3rd param "DetectRed" must be this filename
